src/Microsoft.AspNet.Identity/IdentityServiceCollectionExtensions.cs

```diff
@@ -83,18 +83,19 @@ public static IServiceCollection ConfigureIdentityApplicationCookie(this IServic
             services.AddAuthentication();
 
             // Identity services
-            services.TryAdd(ServiceDescriptor.Scoped<IUserValidator<TUser>, UserValidator<TUser>>());
-            services.TryAdd(ServiceDescriptor.Scoped<IPasswordValidator<TUser>, PasswordValidator<TUser>>());
-            services.TryAdd(ServiceDescriptor.Scoped<IPasswordHasher<TUser>, PasswordHasher<TUser>>());
-            services.TryAdd(ServiceDescriptor.Scoped<ILookupNormalizer, UpperInvariantLookupNormalizer>());
-            services.TryAdd(ServiceDescriptor.Scoped<IRoleValidator<TRole>, RoleValidator<TRole>>());
+            services.TryAddSingleton<IdentityMarkerService>();
```

Comment:
I think it would be slightly better to make it a transient so that it is not parked in memory forever :smile: